src/compiler/parser.ts

```diff
@@ -1858,7 +1861,36 @@ module ts {
         function parseParameterInitializer() {
             return parseInitializer(/*inParameter*/ true);
         }
-
+        
+        function nextTokenIsIsKeyword(): boolean {
+            nextToken();
+            return token === SyntaxKind.IsKeyword;
+        }
+        
+        function parseTypeGuardType(parameterList?: NodeArray<ParameterDeclaration>): TypeGuardTypeNode {
+            let node = <TypeGuardTypeNode>createNode(SyntaxKind.TypeGuardType);
+            if (token === SyntaxKind.Identifier) {
+                node.target = createIdentifier(true);
+            }
+            else if (token === SyntaxKind.ThisKeyword) {
+                nextToken();
+            }
+            if (parseExpected(SyntaxKind.IsKeyword)) {
+                if (token === SyntaxKind.OpenBraceToken) {

```

Comment:
I've tried it before, I don't know why I get this error though:
```
    function isMissingTypeInTypeGuardType(x): x is {
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
!!! error TS2391: Function implementation is missing or not immediately following the declaration.
        return true;
        ~~~~~~
!!! error TS1131: Property or signature expected.
    }
    ~
!!! error TS1128: Declaration or statement expected.
```
Instead of:
```
    function isMissingTypeInTypeGuardType(x): x is {
                                                   ~
!!! error TS1219: Type declaration expected.
        return true;
    }
```

Though I can admit checking for an open brace is not so good. Any good alternatives?